crates/cheatcodes/src/script.rs

```diff
@@ -29,6 +32,57 @@ impl Cheatcode for broadcast_2Call {
     }
 }
 
+impl Cheatcode for createDelegationCall {
+    fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
+        let Self { implementation, nonce } = self;
+        let auth = Authorization {
+            address: *implementation,
+            nonce: *nonce,
+            chain_id: U256::from(ccx.ecx.env.cfg.chain_id),
+        };
+        let hash = auth.signature_hash();
+        Ok(hash.to_vec())
+    }
+}
+
+impl Cheatcode for attachDelegationCall {
+    // @todo - change this to apply_full?
+    fn apply_stateful(&self, ccx: &mut CheatsCtxt) -> Result {
+        let Self { implementation, nonce, v, r, s } = self;
+        let auth = Authorization {
+            address: *implementation,
+            nonce: *nonce,
+            chain_id: U256::from(ccx.ecx.env.cfg.chain_id),
+        };
+        let addr = auth.address;
+        let sig = Signature::from_rs_and_parity(
+            U256::try_from(*r).unwrap(),
+            U256::try_from(*s).unwrap(),
+            alloy_primitives::Parity::from(*v == 1),
+        )?;
+        let signed_auth = auth.into_signed(sig);
+        // store in CheatcodeState
+        ccx.state.delegations.insert(addr, signed_auth);
+        Ok(Default::default())
+    }
+}
+
+impl Cheatcode for signDelegationCall {
+    fn apply_stateful(&self, _: &mut CheatsCtxt) -> Result {
```

Comment:
isn't this same as `vm.sign`? I think we could add a cheat which would both create and sign delegation in a single call

eventually once we have hardware wallets support we could also enable this cheat for more wallets than just raw keys

Reply:
Good call, I've combined `createDelegation` into `signDelegation`.